loop.go

```diff
@@ -15,6 +16,7 @@ var ErrorWaitTime time.Duration = time.Duration(10 * time.Second)
 var metricPrefix string
 
 var statQueue chan Message
+var wg sync.WaitGroup
```

Comment:
I personally don't like using global wait groups since in the future, if this is used incorrectly, it might prevent the intended execution flow. But if there is no better alternative, I guess this is okay. May be call blocking flush for shutdown and use wait group inside that method to prevent returning from blocking flush before shutting down..? There might be a better way to do this, but this is all I can think of right now.